pkg/util/imagestreamtagmapper/imagestreamtagmapper.go

```diff
@@ -43,7 +43,9 @@ func (m *imagestreamtagmapper) Update(e event.UpdateEvent, q workqueue.RateLimit
 	}
 
 	for _, newTag := range newStream.Status.Tags {
-		if namedTagEventListHasElement(oldStream.Status.Tags, newTag) {
+		if namedTagEventListHasElement(oldStream.Status.Tags, newTag) &&
+			//generate requests for the isTags when DeletionTimestamp is defined
+			!(oldStream.DeletionTimestamp == nil && newStream.DeletionTimestamp != nil) {
```

Comment:
Can you change this to:
```
isDeleted := oldStream.DeletionTimestamp == nil && newStream.DeletionTimestamp != nil
for _, newTag := range newStream.Status.Tags {
  if !isDeleted && namedTagEventListHasElement(oldStream.Status.Tags, newTag) {
    continue
  }
}
```

The current multiline if header is kinda annoying to read